executor/include/execution/trait/is_executor.hpp

```diff
@@ -4,24 +4,31 @@
 
 #pragma once
 
+#include <execution/executor/base_executor.hpp>
 #include <experimental/type_traits>
 #include <type_traits>
 
 namespace execution {
 
-template <class Executor, typename = void>
+template <class Executor, typename Function = void (*)(), typename = void>
 struct is_executor : std::false_type {};
 
-constexpr const auto noop = [] {};
+namespace detail {
+const auto noop = [] {};
+}
 
-template <class Executor>
+template <typename Executor, typename Function>
 struct is_executor<
-    Executor,
+    Executor, Function,
     std::enable_if_t<
-        std::is_same_v<void, decltype(std::declval<Executor>().execute(noop))>,
-        void>> : std::true_type {};
+        std::is_copy_constructible_v<
+            std::remove_cv_t<std::remove_reference_t<Executor>>> &&
+            std::is_same_v<
+                decltype(std::declval<Executor>().execute(detail::noop)), void>,
+        decltype(std::declval<Executor&>() == std::declval<Executor&>(),
+                 (void)0)>> : std::true_type {};
```

Comment:
I'm on this party as well. Can you explain in detail what is happening in these two lines.

Reply:
https://github.com/shrijitsingh99/unified-executors/pull/7#discussion_r448564245